File: Python/Binary Tree Inorder Traversal.py
# Definition for a  binary tree node
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    # @param root, a tree node
    # @return a list of integers
    def inorderTraversal(self, root):
		stack = []
		result = []
		current = root
		if current == None:
			return result
		while ((stack != None) or (current != None)):
			if current != None:
				stack.append(current)
				current = current.left
			else:
				current = stack.pop()
				result.append(current.val)
				current = current.right
		return result

File: Java/Binary Tree Inorder Traversal.java
class Binary Tree Inorder Traversal{
	public ArrayList<Integer> inorderTraversal(TreeNode root) {
        ArrayList<Integer> res = new ArrayList<Integer>();
        if(root == null)
        	return res;
        LinkedList<TreeNode> stack = new LinkedList<TreeNode>();
        while(!stack.isEmpty() || root != null){
        	if(root!=null){
        		stack.push(root);
        		root = root.left;
        	}
        	else{
        		root = stack.pop();
        		res.add(root.val);
        		root = root.right;
        	}
        }
        return res;
    }
}